zcash_client_backend/src/fees/common.rs

```diff
@@ -353,35 +424,96 @@ where
             // Case 3b or 3c.
             let proposed_change =
                 (total_in - total_out_plus_fee_with_change).expect("checked above");
+
+            // We obtain a split count based on the total number of notes of sufficient size
+            // available in the wallet, irrespective of pool. If we don't have any wallet metadata
+            // available, we fall back to generating a single change output.
+            let split_count = wallet_meta.map_or(NonZeroUsize::MIN, |wm| {
+                cfg.split_policy
+                    .split_count(wm.total_note_count(), proposed_change)
+            });
+            let per_output_change = proposed_change.div_with_remainder(
+                NonZeroU64::new(
+                    u64::try_from(usize::from(split_count)).expect("usize fits into u64"),
+                )
+                .unwrap(),
+            );
```

Comment:
If `proposed_change` is not a multiple of `split_count`, this will create slightly less change than expected, presumably leaking the remainder publically in the fee. The remainder then gives a lower bound on `split_count`, which should have been private (if the adversary is uncertain about how many real outputs there are).